helpers/parse_args.go

```diff
@@ -12,17 +12,25 @@ import (
 	"os"
 	"strings"
 
+	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/subosito/gotenv"
 )
 
 var (
-	resourceGroupNamePrefix  string
-	resourceGroupName        string
-	location                 string
-	subscriptionID           string
-	servicePrincipalObjectID string
-	keepResources            bool
-	deviceFlow               bool
+	resourceGroupNamePrefix   string
+	resourceGroupName         string
+	location                  string
+	subscriptionID            string
+	servicePrincipalObjectID  string
+	keepResources             bool
+	deviceFlow                bool
+	armEndpointString         string
+	activeDirectoryResourceID string
+	storageEndpointSuffix     string
+	activeDirectoryEndpoint   string
```

Comment:
Instead of storing a few endpoints, I'd rather store one [Environment](https://github.com/Azure/go-autorest/blob/05cc91b23cb791df3737aa915d76c76d216ea732/autorest/azure/environments.go#L37).

You can still populate its fields from command line arguments, if that's what's easiest. :)

One of your colleagues (@radhikagupta5) added some nifty methods to try to automatically populate Stack Environments:
- https://github.com/Azure/go-autorest/blob/05cc91b23cb791df3737aa915d76c76d216ea732/autorest/azure/environments.go#L154
- https://github.com/Azure/go-autorest/blob/05cc91b23cb791df3737aa915d76c76d216ea732/autorest/azure/environments.go#L177

Reply:
These variables are removed